staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go

```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
 }
 
 // DevInit initialize the device
-func (d *DevPanel) DevInit(cfg *config.Config) error {
-	devs := make(map[string]*common.DeviceInstance)
-
-	switch cfg.DevInit.Mode {
-	//case common.DevInitModeConfigmap:
-	//	if err := parse.Parse(cfg.DevInit.Configmap, devs, d.models, d.protocols); err != nil {
-	//		return err
-	//	}
-	case common.DevInitModeRegister:
-		if err := parse.ParseByUsingRegister(devs, d.models, d.protocols); err != nil {
+func (d *DevPanel) DevInit(deviceList []*dmiapi.Device, deviceModelList []*dmiapi.DeviceModel) error {
+	if len(deviceList) == 0 || len(deviceModelList) == 0 {
+		return ErrEmptyData
+	}
+
+	for _, model := range deviceModelList {
```

Comment:
Foreach pointer slice with golang version <1.20 can be problematic, you can use:
```golang
for i := range deviceModelList {
    model := deviceModelList[i]
    ...
}
```